cocos/asset/assets/material.ts

```diff
@@ -448,8 +448,9 @@ export class Material extends Asset {
                 this._passes.forEach((pass, i): void => {
                     let props = this._props[i];
                     if (!props) { props = this._props[i] = {}; }
+                    Object.assign(props, this._effectAsset!.props[i]);
```

Comment:
There seem to be two problems:
1. after assign, the data of effect will be saved on material
2. after assign, if the effect and material have the same attribute, the attribute in material will be overwritten, which is not as expected.